clients/venice-thin-client/src/main/java/com/linkedin/venice/client/schema/RouterBasedStoreSchemaFetcher.java

```diff
@@ -74,6 +75,20 @@ public Schema getLatestValueSchema() {
     }
   }
 
+  @Override
+  public Schema getSupersetSchema() {
```

Comment:
I think replacing `getLatestValueSchema` would be incorrect

We can modify `UpdateBuilderImpl` to fetch superset schema on it's own and that would be a better change. I can make this change. I will have to add a similar implementation for online producer any way

Reply:
I believe after Nisarg's previous change, everyone on UpdateBuilder should ideally be using superset schema so it will be no data loss unless there is any special case...

Reply:
I need to double check with Samza team. It seems like in the current implementation, they still need the schema in the internal pipeline for the KVCoder thing....

Reply:
> I think replacing `getLatestValueSchema` would be incorrect
> 
> We can modify `UpdateBuilderImpl` to fetch superset schema on it's own and that would be a better change. I can make this change. I will have to add a similar implementation for online producer any way

@nisargthakkar 
Can you elaborate why it is incorrect to replace the implementation?

Reply:
I was thinking we'd need a `UpdateBuilderFactory` and the constructor for that will take `StoreSchemaFetcher` as an argument

Reply:
> Can you elaborate why it is incorrect to replace the implementation?

The superset schema might not be the latest value schema. The users will see different latest schemas when using `StoreClient#getLatestValueSchema` vs `StoreSchemaFetcher#getLatestValueSchema`.

The callers could see different schemas from thin-client reads (uses latest id) vs what was fetched from `StoreSchemaFetcher#getLatestValueSchema` (superset schema)

Reply:
Well, today, Venice store client (at least thin client/DVC) will always try to use the superset schema to decode the payload if present.
Updating the implementation of `StoreSchemaFetcher` makes them more aligned:
https://github.com/linkedin/venice/blob/main/clients/venice-thin-client/src/main/java/com/linkedin/venice/client/schema/RouterBackedSchemaReader.java#L271